feat: add C implementation for `math/base/special/roundsd`

ShabiShett07 opened the pull request:
Progresses #649 .

## Description

> What is the purpose of this pull request?

This pull request:

-   progresses #649 by adding math/base/special/roundsds

## Related Issues

> Does this pull request have any related issues?

-progresses #649 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

ShabiShett07 committed:
feat: add math/base/special/roundsd

ShabiShett07 commented:
/stdlib check-files

ShabiShett07 commented:
/stdlib update-copyright-years

stdlib-bot committed:
chore: update copyright years

ShabiShett07 commented:
/stdlib lint-autofix

stdlib-bot committed:
fix: resolve lint errors

Planeshifter commented:
<!-- stdlib-bot-tracking-issue-closure-check -->
:warning: **Tracking Issue Closure Warning** :warning:

I noticed your PR description contains closing keywords ("Resolves", "Closes", or "Fixes") referencing a "Tracking Issue".

**Why this matters:**
Tracking issues should typically remain open until all related sub-issues are completed. GitHub automatically closes issues with such closing keywords when the PR is merged.

**Required action:**
Use "Progresses:" or "Ref:" instead to reference the tracking issue without automatically closing it.

Thank you for your contribution to the project!

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/roundsd/lib/native.js:
```diff
@@ -0,0 +1,60 @@
+* @private
+* @param {number} x - input value
+* @param {PositiveInteger} n - number of significant figures
+* @param {PositiveInteger} [b=10] - base
```
```suggestion
* @param {PositiveInteger} b - base
```

Base won't be optional in the C implementation.

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/roundsd/src/main.c:
```diff
@@ -0,0 +1,102 @@
+* @param x - input value
+* @param n - number of significant figures
+* @param b=10 - base
+* @returns {number} rounded value
```
```suggestion
* @param x       input value
* @param n       number of significant figures
* @param b       base
* @returns       rounded value
```

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/roundsd/src/main.c:
```diff
@@ -0,0 +1,102 @@
+* @returns {number} rounded value
+*
+* @example
+* double v = roundsd( 3.141592653589793, 3, 10 );
```
```suggestion
* double v = stdlib_base_roundsd( 3.141592653589793, 3, 10 );
```

Applies here and below.

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/roundsd/src/main.c:
```diff
@@ -0,0 +1,102 @@
+		stdlib_base_is_infinite( n )
+	) {
+		return 0.0/0.0; //NaN
+	}
```
```suggestion
	if ( stdlib_base_is_nan( x ) || n < 1 ) {
		return 0.0/0.0; //NaN
	}
```

No `NaN` or infinity checks for integer values.

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/roundsd/src/main.c:
```diff
@@ -0,0 +1,102 @@
+	    stdlib_base_is_nan( n ) ||
+		n < 1 ||
+		stdlib_base_is_infinite( n )
+	) {
```
```suggestion
	double y;
	
	if ( stdlib_base_is_nan( x ) || n < 1 ) {
```

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/roundsd/src/main.c:
```diff
@@ -0,0 +1,102 @@
+		exp = stdlib_base_ln( stdlib_base_abs(x) ) / stdlib_base_ln( b );
+	}
+	exp = stdlib_base_floor( exp - n + 1.0 );
+	s = stdlib_base_pow( b, abs( exp ) );
```
```suggestion
	s = stdlib_base_pow( b, stdlib_base_abs( exp ) );
```

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/roundsd/test/test.native.js:
```diff
@@ -0,0 +1,257 @@
+
+// TESTS //
+
+tape( 'main export is a function', function test( t ) {
```
```suggestion
tape( 'main export is a function', opts, function test( t ) {
```

Missing `opts`. Applies here and below.

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/roundsd/test/test.native.js:
```diff
@@ -0,0 +1,257 @@
+	t.strictEqual( isnan( v ), true, 'returns NaN' );
+
+	v = roundsd( NaN, NaN );
+	t.strictEqual( isnan( v ), true, 'returns NaN' );
```
These are not needed because the function needs 3 parameters.

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/roundsd/test/test.native.js:
```diff
@@ -0,0 +1,257 @@
+var isnan = require( '@stdlib/math/base/assert/is-nan' );
+var isNegativeZero = require( '@stdlib/math/base/assert/is-negative-zero' );
+var isPositiveZero = require( '@stdlib/math/base/assert/is-positive-zero' );
+var roundsd = require( './../lib' );
```
This should be refactored to use proper logic for the native tests. Like:

```suggestion
var tryRequire = require( '@stdlib/utils/try-require' );


// VARIABLES //

var roundsd = tryRequire( resolve( __dirname, './../lib/native.js' ) );
var opts = {
	'skip': ( roundsd instanceof Error )
};
```

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/roundsd/test/test.native.js:
```diff
@@ -0,0 +1,257 @@
+	t.strictEqual( isnan( v ), true, 'returns NaN' );
+
+	v = roundsd( NaN, NaN, 10 );
+	t.strictEqual( isnan( v ), true, 'returns NaN' );
```
This is not needed because the second and third arguments being integers, can't be `NaN`.

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/roundsd/test/test.native.js:
```diff
@@ -0,0 +1,257 @@
+	t.strictEqual( isnan( v ), true, 'returns NaN' );
+
+	v = roundsd( 3.14, 2, NaN );
+	t.strictEqual( isnan( v ), true, 'returns NaN' );
```
Same comment for these.

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/roundsd/test/test.native.js:
```diff
@@ -0,0 +1,257 @@
+	t.strictEqual( isnan( v ), true, 'returns NaN' );
+
+	t.end();
+});
```
This is not needed for the same reasons above.

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/roundsd/test/test.native.js:
```diff
@@ -0,0 +1,257 @@
+	t.strictEqual( isnan( v ), true, 'returns NaN' );
+
+	v = roundsd( PI, -1 );
+	t.strictEqual( isnan( v ), true, 'returns NaN' );
```
```suggestion
	v = roundsd( PI, 0, 10 );
	t.strictEqual( isnan( v ), true, 'returns NaN' );

	v = roundsd( PI, -1, 10 );
	t.strictEqual( isnan( v ), true, 'returns NaN' );
```

All of your tests should be refactored to have 3 arguments for `roundsd`.

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/roundsd/test/test.native.js:
```diff
@@ -0,0 +1,257 @@
+});
+
+tape( 'the function returns `+infinity` if provided `+infinity`', function test( t ) {
+	var v = roundsd( PINF, 5 );
```
```suggestion
	var v = roundsd( PINF, 5, 10 );
```

Same comment as above.

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/roundsd/benchmark/benchmark.native.js:
```diff
@@ -0,0 +1,62 @@
+
+	b.tic();
+	for ( i = 0; i < b.iterations; i++ ) {
+		x = ( randu()*1000.0 ) - 500.0;
```
Can you refactor it to use `uniform` instead, as other JS benchmarks?

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/roundsd/benchmark/c/benchmark.c:
```diff
@@ -0,0 +1,137 @@
+/**
```
This file will be inside `c/native` directory.

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/roundsd/benchmark/c/benchmark.c:
```diff
@@ -0,0 +1,137 @@
+	t = tic();
+	for ( i = 0; i < ITERATIONS; i++ ) {
+		v = ( 1000.0*rand_double() ) - 500.0;
+        n = ( 1000.0*rand_double() ) - 500.0;
+        b = ( 1000.0*rand_double() ) - 500.0;
```
`n` and `b` do not `double` data type.

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/roundsd/benchmark/c/benchmark.c:
```diff
@@ -0,0 +1,137 @@
+		v = ( 1000.0*rand_double() ) - 500.0;
+        n = ( 1000.0*rand_double() ) - 500.0;
+        b = ( 1000.0*rand_double() ) - 500.0;
+		y = stdlib_base_roundsd( v, n, b );
```
```suggestion
		y = stdlib_base_roundsd( v, 5, 10 );
```

You can change this instead.

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/roundsd/README.md:
```diff
@@ -87,36 +87,106 @@ for ( i = 0; i < 100; i++ ) {
-[@stdlib/math/base/special/ceilsd]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/ceilsd
+#### stdlib_base_roundsd( x, n, b )
+
+Rounds a numeric value to the nearest number with \\(n\\) significant figures.
```
```suggestion
Rounds a `numeric` value to the nearest `number` with `n` significant figures.
```

Same description as above.

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/roundsd/README.md:
````diff
@@ -87,36 +87,106 @@ for ( i = 0; i < 100; i++ ) {
+
+```c
+ double v = roundsd( 3.141592653589793, 3, 10 );
+ // returns 3.14
````
```suggestion
double v = stdlib_base_roundsd( 3.141592653589793, 3, 10 );
// returns 3.14
```

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/roundsd/README.md:
```diff
@@ -87,36 +87,106 @@ for ( i = 0; i < 100; i++ ) {
+    double y;
+    int i;
+    for ( i = 0; i < 4; i++ ) {
+        y = stdlib_base_roundsd( x[ i ], -2 );
```
```suggestion
        y = stdlib_base_roundsd( x[ i ], -2, 10 );
```

anandkaranubc reviewed (changes requested):
Hi @ShabiShett07, thanks for your work on this! I've left a number of comments on the PR. There are quite a few areas that need improvement. When you get a chance, please take a look and let me know if anything is unclear. Happy to help clarify!

kgryte commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into feature/roundsd

ShabiShett07 committed:
fix: jsdoc

Co-authored-by: Karan Anand <119553199+anandkaranubc@users.noreply.github.com>
Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
fix: jsdoc to doxygen

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
chore: update doxygen

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
chore: add suggestions

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: missing_dependencies
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: missing_dependencies
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

ShabiShett07 commented:
Thanks a lot @anandkaranubc for spending your time in this PR and I am really sorry that you had to make a plenty of suggestions . I have tried to implement all your suggestions.

ShabiShett07 committed:
fix: lint error

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

ShabiShett07 committed:
fix: lint error

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: missing_dependencies
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

ShabiShett07 committed:
remove: unused variables

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/roundsd/benchmark/benchmark.native.js:
```diff
@@ -0,0 +1,63 @@
+	for ( i = 0; i < b.iterations; i++ ) {
+		y = roundsd( x[ i%x.length ], 2, 2 );
+		if ( isnan( y ) ) {
+			console.log('Generated value:', x);
```
We don't need these console statements.

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/roundsd/benchmark/benchmark.native.js:
```diff
@@ -0,0 +1,63 @@
+	}
+	b.toc();
+	if ( isnan( y ) ) {
+		console.log('Generated value:', x);
```
Same comment

ShabiShett07 committed:
chore: remove console

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 commented:
@anandkaranubc, I have resolved the console ones too.

anandkaranubc commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into feature/roundsd

anandkaranubc committed:
test: update test messages

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
bench: move random number generation outside the benchmarking loops

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
docs: fix C script for generating examples

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc reviewed (dismissed):
Ready to land! 🚀
Thanks @ShabiShett07!

kgryte commented on lib/node_modules/@stdlib/math/base/special/roundsd/README.md:
```diff
@@ -85,36 +85,106 @@ logEachMap( 'x: %0.4f. y: %d. Rounded: %0.4f.', x, 5, roundsd );
+
+## C APIs
 
--   <span class="package-name">[`@stdlib/math/base/special/ceilsd`][@stdlib/math/base/special/ceilsd]</span><span class="delimiter">: </span><span class="description">round a numeric value to the nearest number toward positive infinity with N significant figures.</span>
```
While this will get reinserted through automation, in general, for existing packages, we shouldn't be manually removing the list of related packages. There is not a reason for doing so.

kgryte commented on lib/node_modules/@stdlib/math/base/special/roundsd/README.md:
```diff
@@ -85,36 +85,106 @@ logEachMap( 'x: %0.4f. y: %d. Rounded: %0.4f.', x, 5, roundsd );
+
+#### stdlib_base_roundsd( x, n, b )
+
+Rounds a `numeric` value to the nearest `number` with `n` significant figures.
```
```suggestion
Rounds a double-precision floating-point number to the nearest value with `n` significant figures.
```

kgryte replied:
It's likely that we should update the description elsewhere throughout this package.

kgryte commented on lib/node_modules/@stdlib/math/base/special/roundsd/include/stdlib/math/base/special/roundsd.h:
```diff
@@ -0,0 +1,40 @@
+#endif
+
+/**
+* Rounds a numeric value to the nearest number with \\(n\\) significant figures.
```
@ShabiShett07 Mind updating the description with similar edits to that proposed for the README? You can leave the `\\(n\\)` TeX markup here.

kgryte commented on lib/node_modules/@stdlib/math/base/special/roundsd/lib/native.js:
```diff
@@ -0,0 +1,60 @@
+// MAIN //
+
+/**
+* Rounds a numeric value to the nearest number with \\(n\\) significant figures.
```
Same comment here and elsewhere.

kgryte commented on lib/node_modules/@stdlib/math/base/special/roundsd/manifest.json:
```diff
@@ -0,0 +1,102 @@
+{
+    "options": {
```
This file should be dedented by two spaces to ensure consistent 2-space indentation.

kgryte commented on lib/node_modules/@stdlib/math/base/special/roundsd/src/main.c:
```diff
@@ -0,0 +1,86 @@
+* limitations under the License.
+*/
+
+
```
```suggestion
```

kgryte commented on lib/node_modules/@stdlib/math/base/special/roundsd/src/main.c:
```diff
@@ -0,0 +1,86 @@
+// MAIN //
+
+/**
+* Rounds a numeric value to the nearest number with \\(n\\) significant figures.
```
Same comment.

kgryte commented on lib/node_modules/@stdlib/math/base/special/roundsd/src/main.c:
```diff
@@ -0,0 +1,86 @@
+* @param x   input value
+* @param n   number of significant figures
+* @param b   base
+* @returns   rounded value
```
```suggestion
* @return    rounded value
```

We use slightly different tags in C docs.

kgryte commented on lib/node_modules/@stdlib/math/base/special/roundsd/src/main.c:
```diff
@@ -0,0 +1,86 @@
+	}
+	if ( b == 10 ) {
+		exp = stdlib_base_log10( stdlib_base_abs( x ) );
+	}
+	else if ( b == 2 ) {
+		exp = stdlib_base_float64_exponent( stdlib_base_abs( x ) );
+	}
+	else {
```
```suggestion
	} else if ( b == 2 ) {
		exp = stdlib_base_float64_exponent( stdlib_base_abs( x ) );
	} else {
```

Linebreaks are not necessary here.

kgryte reviewed (changes requested):
Other than some minor nitpicks, this is coming along.

ShabiShett07 committed:
chore: add minor changes

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: missing_dependencies
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

ShabiShett07 commented:
@kgryte, thank you for the corrections and I have resolved all the conversations

kgryte committed:
docs: update description

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/math/base/special/roundsd/README.md:
```diff
@@ -20,7 +20,7 @@ limitations under the License.
 # roundsd
 
-> Round a numeric value to the nearest number with `n` significant figures.
+> Rounds a double-precision floating-point number to the nearest value with `n` significant figures.
```
```suggestion
> Round a double-precision floating-point number to the nearest value with `n` significant figures.
```

kgryte committed:
docs: fix description

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/math/base/special/roundsd/docs/repl.txt:
```diff
@@ -1,6 +1,7 @@
 
 {{alias}}( x, n[, b] )
-    Rounds a numeric value to the nearest number with `n` significant figures.
+    Rounds a double-precision floating-point number to the nearest
+    value with `n` significant figures.
```
```suggestion
    Rounds a double-precision floating-point number to the nearest value with
    `n` significant figures.
```

kgryte committed:
docs: fix line wrapping

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/math/base/special/roundsd/include/stdlib/math/base/special/roundsd.h:
```diff
@@ -0,0 +1,40 @@
+#endif
+
+/**
+* Rounds a double-precision floating-point number to the nearest value with \\(n\\) significant figures.
```
```suggestion
* Rounds a double-precision floating-point number to the nearest value with `n` significant figures.
```

kgryte committed:
style: remove TeX and use backticks to be consistent with elsewhere

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/math/base/special/roundsd/package.json:
```diff
@@ -1,7 +1,7 @@
   "name": "@stdlib/math/base/special/roundsd",
   "version": "0.0.0",
-  "description": "Round a numeric value to the nearest number with N significant figures.",
+  "description": "Rounds a double-precision floating-point number to the nearest value with N significant figures.",
```
```suggestion
  "description": "Round a double-precision floating-point number to the nearest value with N significant figures.",
```

kgryte committed:
docs: fix description

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/math/base/special/roundsd/src/main.c:
```diff
@@ -0,0 +1,83 @@
+// MAIN //
+
+/**
+* Rounds a double-precision floating-point number to the nearest value with \\(n\\) significant figures.
```
```suggestion
* Rounds a double-precision floating-point number to the nearest value with `n` significant figures.
```

kgryte committed:
style: use consistent markup

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/math/base/special/roundsd/src/main.c:
```diff
@@ -0,0 +1,83 @@
+	} else if ( b == 2 ) {
+		exp = stdlib_base_float64_exponent( stdlib_base_abs( x ) );
+	} else {
+		exp = stdlib_base_ln( stdlib_base_abs(x) ) / stdlib_base_ln( b );
```
```suggestion
		exp = stdlib_base_ln( stdlib_base_abs( x ) ) / stdlib_base_ln( b );
```

kgryte committed:
style: use consistent spacing

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/math/base/special/roundsd/src/main.c:
```diff
@@ -0,0 +1,83 @@
+	double y;
+
+	if ( stdlib_base_is_nan( x ) || n < 1 ) {
+		return 0.0/0.0; //NaN
```
```suggestion
		return 0.0/0.0; // NaN
```

kgryte committed:
style: add missing space

Signed-off-by: Athan <kgryte@gmail.com>

kgryte reviewed (approved):
LGTM after clean-up.

@ShabiShett07 General comment: you'll set yourself up better for long-term success contributing to stdlib if you can pay more attention to detail. In this case, even after updating the descriptions, you ended up diverging from stdlib conventions. When in doubt, study other packages and mimic what you see there. Occasionally, conventions do change, but this is not a frequent occurrence. Currently, a lot of your PRs require a fair amount of clean-up. Would be good to minimize that overhead.

ShabiShett07 commented:
> LGTM after clean-up.
> 
> @ShabiShett07 General comment: you'll set yourself up better for long-term success contributing to stdlib if you can pay more attention to detail. In this case, even after updating the descriptions, you ended up diverging from stdlib conventions. When in doubt, study other packages and mimic what you see there. Occasionally, conventions do change, but this is not a frequent occurrence. Currently, a lot of your PRs require a fair amount of clean-up. Would be good to minimize that overhead.

I am really sorry for that, actually didn't give much time to this PR, as I have done this in my initial times, also I think yes, I need to improve a lot in detailing, but thank you for letting me know y mistakes
